### DISCUSSION TITLE
State of AoT in Maui?

### DISCUSSION THREAD

**Discussion by DDHSchmidt** [dotnet/maui#15682] [Q&A]
I am testing out .Net 8 Preview 5 today and one of those tests concerned if I could get a release build with Ahead-of-time compilation built.

I tried for iOS and Windows and unsurprisingly it ended in crashes with obscure error messages for our MAUI app.
I was about to file new issues, but then I thought:
"*Is* Maui even supposed to use AoT (yet)?"

I went back to all the announcements of the .Net 8 preview versions and sure enough they tout AoT as the [next best thing besides sliced bread](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-4/#native-aot-improvements).
But then I checked with all the [announcements](https://devblogs.microsoft.com/dotnet/announcing-dotnet-maui-in-dotnet-8-preview-3/) of the [Maui 8 preview versions](https://devblogs.microsoft.com/dotnet/announcing-dotnet-maui-in-dotnet-8-preview-5/) and AoT is not mentioned once in any of them.

The [documentation page for AoT](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net8plus) also mentions some general limitatoins but nothing about Maui itself.

What I'm trying to figure out is if Maui is *supposed* to benefit from AoT at all? And if yes: Is that planned for .Net 8 already?

**Comment by xtuzy** [comment]
Maui use mono on iOS and Android, current NativeAOT support coreclr, mono has its own AOT, current mono's AOT is enough to develop iOS / Android app
